src/Symfony/Component/Form/ChoiceList/Loader/ExtraLazyChoiceLoader.php

```diff
@@ -0,0 +1,54 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Form\ChoiceList\Loader;
+
+use Symfony\Component\Form\ChoiceList\ArrayChoiceList;
+use Symfony\Component\Form\ChoiceList\ChoiceListInterface;
+
+/**
+ * A choice loader that loads its choices and values lazily, only when necessary.
+ *
+ * @author Yonel Ceruto <yonelceruto@gmail.com>
+ */
+class ExtraLazyChoiceLoader implements ChoiceLoaderInterface
```

Comment:
Why is it called "extra" lazy?

Can't we have laziness by default also?

Reply:
It's already lazy because there is a wrapped loader performing the main function, and the "extra" refers to the `loadChoiceList()` strategy, which caches the choices in memory but changes them on demand. I thought of this name by looking at the `extra_lazy` strategy of Doctrine associations.

Naming things is always so hard. 😅

Reply:
i understand "extra" conveys the technical aspect, but it doesnt convey to users what it's about.

shouldnt we simply call it `autocompletes=true`

Reply:
otherwise, i'd go with just `lazy`, since that option currently does not exists

Reply:
`lazy` works for me

Reply:
changed to `choice_lazy` to be more specific